bluetooth_mesh/mesh.py

```diff
@@ -141,8 +144,50 @@ def transport_pdu(self, application_key, seq, iv_index):
             yield seq, bitstring.pack('uint:1, uint:1, uint:6, bytes',
                                       seg, akf, aid, upper_transport_pdu).bytes
 
+    def __reassembly(self):
```

Comment:
I think my initial idea to use generators for SAR might work for segmentation, but royally sucks in case of reassembly...
Both this, and proxy SAR should be refactored.

Just a comment, nothing about your PR ;)

Reply:
this part is not tested yet. 
yes, we do need a better way to handle all the cases, message loss, wrong sequence, corrupted content, conflicts due to different message source and so on.